lib/block-supports/layout.php

```diff
@@ -241,15 +241,18 @@ function gutenberg_render_layout_support_flag( $block_content, $block ) {
 	$fallback_gap_value = _wp_array_get( $block_type->supports, array( 'spacing', 'blockGap', '__experimentalDefault' ), '0.5em' );
 	$block_spacing      = _wp_array_get( $block, array( 'attrs', 'style', 'spacing' ), null );
 
-	// If a block's block.json skips serialization for spacing or spacing.blockGap,
-	// don't apply the user-defined value to the styles.
-	$should_skip_gap_serialization = gutenberg_should_skip_block_supports_serialization( $block_type, 'spacing', 'blockGap' );
-	$style                         = gutenberg_get_layout_style( ".$block_classname.$container_class", $used_layout, $has_block_gap_support, $gap_value, $should_skip_gap_serialization, $fallback_gap_value, $block_spacing );
-
-	// Only add container class and enqueue block support styles if unique styles were generated.
-	if ( ! empty( $style ) ) {
-		$class_names[] = $container_class;
-		wp_enqueue_block_support_styles( $style );
+	// Only generate Layout styles if the theme has not opted-out.
+	if ( ! current_theme_supports( 'disable-layout-styles' ) ) {
```

Comment:
There's also a bunch of layout-related logic towards the top of this function. Would it make sense to return early instead of adding this condition here?

Reply:
Thanks for testing! In this case, I don't think we can return early, because we still want the Layout support's semantic classnames to be added to the output which happens at line 260. Part of the feedback that inspired this PR was that there are Classic themes that would like the semantic classnames like layout type and content justification to be available, but not the generated Layout styles themselves.

It does make it a bit of an awkward place for the check to live, though! I suppose we could move it up to line 228 so that this `if` block includes the `$gap_value` declaration and everything below that?

Reply:
Oooh, of course, that makes sense 🤦  Yeah, I guess we could move it up a little bit, so there are no unnecessary calculations performed.

Reply:
I've moved it up a little in https://github.com/WordPress/gutenberg/pull/42544/commits/d77120ea1bae6dfc15bf9b0deee9ee7e00b37d93, which feels a bit more logical to me now, thanks for the suggestion! Also added a comment about the classnames so it's clearer why we're not returning early.

Separately to this PR, now that we've got a fair few classes being generated it might be worth us re-exploring splitting out the classnames logic to a separate function to make it a bit easier to read. We had it in a separate function for a little while but reverted in https://github.com/WordPress/gutenberg/pull/41885 so that we could give it some more thought before committing to a particular approach that we then need to continue to support. But 🤞 we can settle on an approach before 6.1.